pages/learn/develop/dockerfile.md

```diff
@@ -101,6 +101,18 @@ Here are the supported machine names and architectures:
 
 {{> "general/deviceTypeNames"}}
 
+## Multiple Dockerfiles
+
+There are cases where just specifying a Dockerfile template doesn't work, as different systems or architectures may need different configuration and installation files. The solution to this problem is having multiple dockerfiles that are automatically selected based on the device it is going to be used.
+
+When using the command `balena push`, the balena CLI tool looks at all available dockerfiles and builds the appropriate container based on the following order:
```

Comment:
* Instead of _"builds the appropriate container",_ perhaps _"builds the appropriate docker image"._ (Because: a container is run out of a pre-built image; a container is a running instance of an image; an image is built.)
* Re _"the balena CLI tool looks at...", I am almost sure that:

Command | Where Dockerfile resolution and image building takes place
--- | ---
`git push` | All Dockerfiles are sent to the balenaCloud builder, where the appropriate Dockerfile is chosen.
`balena push` | All Dockerfiles are sent to the balenaCloud builder, where the appropriate Dockerfile is chosen.
`balena build` | The CLI chooses chooses the right file. The right file is sent to the specified docker daemon on a balena device or on a development machine (laptop or customer's own server), where the docker images are built.
`balena deploy` | When the `--build` option is given, it behaves like `balena build`. After the images are built, they are uploaded to balenaCloud.

* CLI version 9.14.5 and later is recommended as it fixes relevant issues.